06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js

```diff
@@ -0,0 +1,174 @@
+      testing: [0],
+      soon: [1],
+    });
+  });
+  it('should return true if search term is a string', () => {
+    const words = 'Alice in Wonderland';
+    expect(Object.keys(invertedIndex.searchIndex('words'))).toBeTruthy();
+  });
+  it('Should return correct index of the search term in test.json', () => {
+    expect(invertedIndex.searchIndex('Honey, humans', 'test.json')).toEqual({
+      honey: [0],
+      humans: [1],
+    });
+  });
+});
+
+describe('Tokenize words', () => {
+  it('should return tokenized words for terms', () => {
+    let words = 'It CAN only BE++   GOD';
+    const termsExpected = 'it can only be god';
+    words = invertedIndex.tokenize(words);
+    expect(termsExpected).toEqual(words);
+  });
+  it('should return tokenized words for terms', () => {
+    let token = 'ALICE has GoNE to the maRKEt';
+    const tokenExpected = 'alice has gone to the market';
+    token = invertedIndex.tokenize(token);
+    expect(tokenExpected).toEqual(token);
+  });
+});
+
+describe('Validate File', () => {
+  it('should return truthy if book data is valid',
+    () => {
+      expect(Index.validateFile(work).status).toBeTruthy();
+    });
+  it('should return truthy if book data is valid', () => {
+    expect(Index.validateFile(workBook).status).toBeTruthy();
+  });
+  it('should return validate.msg if file uploaded is valid', () => {
+    expect(Index.validateFile(workBook).msg).toBe('This is a valid File');
+  });
+  it('should return falsy if book is not an array of object literals', () => {
+    expect(Index.validateFile('theDojo').status).toBeFalsy();
+    expect(Index.validateFile(100).status).toBeFalsy();
+  });
+  it('should return validate.msg if file uploaded is empty', () => {
+    expect(Index.validateFile('Dojo').msg).toBe('File is empty upload a file');
+    expect(Index.validateFile(100).msg).toBe('File is empty upload a file');
+  });
+  it('should return validate.msg if file dosent contain title or text', () => {
+    expect(Index.validateFile(empty).msg).toBe('Invalid file content');
+  });
+});
+
+describe('InvertedIndex class, check all methods', () => {
+  it('should check that the class has a createIndex method', () => {
+    expect(typeof invertedIndex.createIndex).toBe('function');
+  });
+
+  it('should check that the class has a validateFile method', () => {
+    expect(typeof Index.validateFile).toBe('function');
+  });
+
+  it('should check that the class has a tokenize method', () => {
+    expect(typeof invertedIndex.tokenize).toBe('function');
+  });
+
+  it('should check that the class has a getIndex method', () => {
+    expect(typeof invertedIndex.getIndex).toBe('function');
+  });
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.searchIndex).toBe('function');
+  });
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.flattenSearch).toBe('function');
+  });
+});
+
```

Comment:
Your tests in general should use "Ubiquitous Language," 

Which is a body of knowledge that can be understood by both the developer and the customer. This ubiquitous language is used to shape and develop the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", function (){

    it("should be able to add 2 numbers together", function (){
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", function (){
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```